planning/behavior_path_planner/autoware_behavior_path_planner/src/planner_manager.cpp

```diff
@@ -135,7 +135,7 @@ BehaviorModuleOutput PlannerManager::run(const std::shared_ptr<PlannerData> & da
   const bool is_any_module_running =
     is_any_approved_module_running || is_any_candidate_module_running_or_idle;
 
-  updateCurrentRouteLanelet(data);
+  updateCurrentRouteLanelet(data, is_any_approved_module_running);
```

Comment:
if `is_any_approved_module_running`, then when the ego path is  on the middle of two lanes while avoiding, ReferencePath can suddenly switch to the next lane. To prevent it, while modules are running, do not call this (resetCurrentRouteLanelet is called when lane_change successfully finished.)